packages/use-contractkit/src/modals/connect.tsx

```diff
@@ -102,137 +116,59 @@ export const ConnectModal: React.FC<ConnectModalProps> = ({
 
     return {
       wallets: _wallets,
-      allScreens: {
-        ..._screens,
-        ..._wallets.reduce((acc, wallet) => {
-          acc[wallet.id] = walletToScreen(wallet);
-          return acc;
-        }, {} as Record<string, React.FC<ConnectorProps>>),
-      },
+      allScreens: _wallets.reduce((acc, wallet) => {
+        acc[wallet.id] = walletToScreen(wallet);
+        return acc;
+      }, _screens),
     };
   }, [screens, hideFromDefaults, additionalWCWallets]);
 
-  const _providers: Record<string, Provider> = useMemo(
-    () => ({
-      ...PROVIDERS,
-      ...wallets.reduce((acc, wallet) => {
-        acc[wallet.id] = walletToProvider(wallet);
-        return acc;
-      }, {} as Record<string, Provider>),
-    }),
-    [wallets]
-  );
-
-  const close = useCallback((): void => {
-    setAdding(null);
-    setShowMore(false);
-    connectionCallback?.(false);
-  }, [connectionCallback]);
+  const providers = useProviders(wallets, sort, search);
 
-  const onSubmit = useCallback(
-    (connector: Connector): void => {
-      setAdding(null);
-      connectionCallback?.(connector);
-    },
-    [connectionCallback]
+  const ProviderElement = adding && allScreens?.[adding];
+  const content = !ProviderElement ? (
```

Comment:
Generally I think it's easier to think of a ternary without the condition being flipped. Instead I would switch the options of the ternary.
`!foo ? notFoo : yesFoo` vs `foo ? yesFoo : notFoo`

Reply:
Agreed.